File: src/main/java/com/box/BoxTester.java
package com.box;

import java.util.ArrayList;
import java.util.List;
import java.util.Scanner;

public class BoxTester {

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Please enter object's length: ");
        float length = scanner.nextFloat();
        System.out.print("Please enter object's width: ");
        float width = scanner.nextFloat();
        System.out.print("Please enter object's height: ");
        float height = scanner.nextFloat();

        System.out.println(box(length, width, height));

    }

    public static String box(float length, float width, float height) {
        Box box3 = new Box3();
        Box box5 = new Box5();

        List<Box> listBox = new ArrayList<>();
        listBox.add(box3);
        listBox.add(box5);

        String validBox = "";
        for (Box box : listBox) {
            if (box.validate(length, width, height)) {
                validBox += box.BoxName();
                break;
            }
        }

        return validBox;
    }

}

File: src/main/kotlin/com/box/BoxTester.kt
package com.box.kotlin

import java.util.*

fun main(args: Array<String>) {
    var scanner = Scanner(System.`in`);
    print("Please enter object's length: ");
    val length = scanner.nextFloat();
    print("Please enter object's width: ");
    val width = scanner.nextFloat();
    print("Please enter object's height: ");
    val height = scanner.nextFloat();

    print(box(length, width, height));
}

fun box(length: Float, width: Float, height: Float): String {
    val box3 = Box3();
    val box5 = Box5();

    var listBox = listOf<Box>(box3, box5);
    var validBox: String = "";
    for (box in listBox) {
        if (box.validBox(length, width, height)) {
            validBox = box.boxName();
            break;
        }
    }
    return validBox;
}
